pkg/hostagent/pods.go

```diff
@@ -386,3 +387,19 @@ func (agent *HostAgent) podDeletedLocked(obj interface{}) {
 	}
 	agent.epDeleted(&u)
 }
+
+func (agent *HostAgent) cniEpDelete(cniKey string) {
+	agent.indexMutex.Lock()
+	defer agent.indexMutex.Unlock()
+	epUuid, ok := agent.cniToPodID[cniKey]
+	if !ok {
+		agent.log.Warnf("cniEpDelete: PodID not found for %s", cniKey)
+		return
+	}
+	delete(agent.cniToPodID, cniKey)
+
+	if _, ok := agent.opflexEps[epUuid]; ok {
+		delete(agent.opflexEps, epUuid)
+		agent.scheduleSyncEps()
+	}
```

Comment:
Pl add a log here and the when we come via cni path and in the original code thats also attempting this delete.